### PR TITLE
Add additional configuration environment variables to CLI help.

### PR DESCRIPTION
**Pull request by ldez**
- add missing DNS provider in the CLI help.
- add additional configuration environment variables to CLI help.
- sort DNS providers.
- fix some wrong names.

Result:
```console
$ ./lego dnshelp
Credentials for DNS providers must be passed through environment variables.

Here is an example bash command using the CloudFlare DNS provider:

  $ CLOUDFLARE_EMAIL=foo@bar.com \
    CLOUDFLARE_API_KEY=b9841238feb177a84330febba8a83208921177bffe733 \
    lego --dns cloudflare --domains www.example.com --email me@bar.com run

Valid providers and their associated credential environment variables:

	acme-dns:	ACME_DNS_API_BASE, ACME_DNS_STORAGE_PATH
	alidns:		ALICLOUD_ACCESS_KEY, ALICLOUD_SECRET_KEY
	auroradns:	AURORA_USER_ID, AURORA_KEY, AURORA_ENDPOINT
	azure:		AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, AZURE_SUBSCRIPTION_ID, AZURE_TENANT_ID, AZURE_RESOURCE_GROUP
	bluecat:	BLUECAT_SERVER_URL, BLUECAT_USER_NAME, BLUECAT_PASSWORD, BLUECAT_CONFIG_NAME, BLUECAT_DNS_VIEW
	cloudflare:	CLOUDFLARE_EMAIL, CLOUDFLARE_API_KEY
	cloudxns:	CLOUDXNS_API_KEY, CLOUDXNS_SECRET_KEY
	digitalocean:	DO_AUTH_TOKEN
	dnsimple:	DNSIMPLE_EMAIL, DNSIMPLE_OAUTH_TOKEN
	dnsmadeeasy:	DNSMADEEASY_API_KEY, DNSMADEEASY_API_SECRET
	dnspod:		DNSPOD_API_KEY
	duckdns:	DUCKDNS_TOKEN
	dyn:		DYN_CUSTOMER_NAME, DYN_USER_NAME, DYN_PASSWORD
	exec:		EXEC_PATH, EXEC_MODE
	exoscale:	EXOSCALE_API_KEY, EXOSCALE_API_SECRET, EXOSCALE_ENDPOINT
	fastdns:	AKAMAI_HOST, AKAMAI_CLIENT_TOKEN, AKAMAI_CLIENT_SECRET, AKAMAI_ACCESS_TOKEN
	gandi:		GANDI_API_KEY
	gandiv5:	GANDIV5_API_KEY
	gcloud:		GCE_PROJECT, GCE_SERVICE_ACCOUNT_FILE
	glesys:		GLESYS_API_USER, GLESYS_API_KEY
	godaddy:	GODADDY_API_KEY, GODADDY_API_SECRET
	hostingde:	HOSTINGDE_API_KEY, HOSTINGDE_ZONE_NAME
	iij:		IIJ_API_ACCESS_KEY, IIJ_API_SECRET_KEY, IIJ_DO_SERVICE_CODE
	lightsail:	AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, DNS_ZONE
	linode:		LINODE_API_KEY
	linodev4:	LINODE_TOKEN
	manual:		none
	namecheap:	NAMECHEAP_API_USER, NAMECHEAP_API_KEY
	namedotcom:	NAMECOM_USERNAME, NAMECOM_API_TOKEN
	netcup:		NETCUP_CUSTOMER_NUMBER, NETCUP_API_KEY, NETCUP_API_PASSWORD
	nifcloud:	NIFCLOUD_ACCESS_KEY_ID, NIFCLOUD_SECRET_ACCESS_KEY
	ns1:		NS1_API_KEY
	otc:		OTC_USER_NAME, OTC_PASSWORD, OTC_PROJECT_NAME, OTC_DOMAIN_NAME, OTC_IDENTITY_ENDPOINT
	ovh:		OVH_ENDPOINT, OVH_APPLICATION_KEY, OVH_APPLICATION_SECRET, OVH_CONSUMER_KEY
	pdns:		PDNS_API_KEY, PDNS_API_URL
	rackspace:	RACKSPACE_USER, RACKSPACE_API_KEY
	rfc2136:	RFC2136_TSIG_KEY, RFC2136_TSIG_SECRET,
			RFC2136_TSIG_ALGORITHM, RFC2136_NAMESERVER
	route53:	AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_REGION, AWS_HOSTED_ZONE_ID
	sakuracloud:	SAKURACLOUD_ACCESS_TOKEN, SAKURACLOUD_ACCESS_TOKEN_SECRET
	vegadns:	SECRET_VEGADNS_KEY, SECRET_VEGADNS_SECRET, VEGADNS_URL
	vultr:		VULTR_API_KEY

Additional configuration environment variables:

	alidns:		ALICLOUD_POLLING_INTERVAL, ALICLOUD_PROPAGATION_TIMEOUT, ALICLOUD_TTL, ALICLOUD_HTTP_TIMEOUT
	auroradns:	AURORA_POLLING_INTERVAL, AURORA_PROPAGATION_TIMEOUT, AURORA_TTL
	azure:		AZURE_POLLING_INTERVAL, AZURE_PROPAGATION_TIMEOUT, AZURE_TTL
	bluecat:	BLUECAT_POLLING_INTERVAL, BLUECAT_PROPAGATION_TIMEOUT, BLUECAT_TTL, BLUECAT_HTTP_TIMEOUT
	cloudflare:	CLOUDFLARE_POLLING_INTERVAL, CLOUDFLARE_PROPAGATION_TIMEOUT, CLOUDFLARE_TTL, CLOUDFLARE_HTTP_TIMEOUT
	cloudxns:	CLOUDXNS_POLLING_INTERVAL, CLOUDXNS_PROPAGATION_TIMEOUT, CLOUDXNS_TTL, CLOUDXNS_HTTP_TIMEOUT
	digitalocean:	DO_POLLING_INTERVAL, DO_PROPAGATION_TIMEOUT, DO_TTL, DO_HTTP_TIMEOUT
	dnsimple:	DNSIMPLE_TTL, DNSIMPLE_POLLING_INTERVAL, DNSIMPLE_PROPAGATION_TIMEOUT
	dnsmadeeasy:	DNSMADEEASY_POLLING_INTERVAL, DNSMADEEASY_PROPAGATION_TIMEOUT, DNSMADEEASY_TTL, DNSMADEEASY_HTTP_TIMEOUT
	dnspod:		DNSPOD_POLLING_INTERVAL, DNSPOD_PROPAGATION_TIMEOUT, DNSPOD_TTL, DNSPOD_HTTP_TIMEOUT
	duckdns:	DUCKDNS_POLLING_INTERVAL, DUCKDNS_PROPAGATION_TIMEOUT, DUCKDNS_HTTP_TIMEOUT
	dyn:		DYN_POLLING_INTERVAL, DYN_PROPAGATION_TIMEOUT, DYN_TTL, DYN_HTTP_TIMEOUT
	exoscale:	EXOSCALE_POLLING_INTERVAL, EXOSCALE_PROPAGATION_TIMEOUT, EXOSCALE_TTL, EXOSCALE_HTTP_TIMEOUT
	fastdns:	AKAMAI_POLLING_INTERVAL, AKAMAI_PROPAGATION_TIMEOUT, AKAMAI_TTL
	gandi:		GANDI_POLLING_INTERVAL, GANDI_PROPAGATION_TIMEOUT, GANDI_TTL, GANDI_HTTP_TIMEOUT
	gandiv5:	GANDIV5_POLLING_INTERVAL, GANDIV5_PROPAGATION_TIMEOUT, GANDIV5_TTL, GANDIV5_HTTP_TIMEOUT
	gcloud:		GCE_POLLING_INTERVAL, GCE_PROPAGATION_TIMEOUT, GCE_TTL
	glesys:		GLESYS_POLLING_INTERVAL, GLESYS_PROPAGATION_TIMEOUT, GLESYS_TTL, GLESYS_HTTP_TIMEOUT
	godaddy:	GODADDY_POLLING_INTERVAL, GODADDY_PROPAGATION_TIMEOUT, GODADDY_TTL, GODADDY_HTTP_TIMEOUT
	hostingde:	HOSTINGDE_POLLING_INTERVAL, HOSTINGDE_PROPAGATION_TIMEOUT, HOSTINGDE_TTL, HOSTINGDE_HTTP_TIMEOUT
	iij:		IIJ_POLLING_INTERVAL, IIJ_PROPAGATION_TIMEOUT, IIJ_TTL
	lightsail:	LIGHTSAIL_POLLING_INTERVAL, LIGHTSAIL_PROPAGATION_TIMEOUT
	linode:		LINODE_POLLING_INTERVAL, LINODE_TTL, LINODE_HTTP_TIMEOUT
	linodev4:	LINODE_POLLING_INTERVAL, LINODE_TTL, LINODE_HTTP_TIMEOUT
	namecheap:	NAMECHEAP_POLLING_INTERVAL, NAMECHEAP_PROPAGATION_TIMEOUT, NAMECHEAP_TTL, NAMECHEAP_HTTP_TIMEOUT
	namedotcom:	NAMECOM_POLLING_INTERVAL, NAMECOM_PROPAGATION_TIMEOUT, NAMECOM_TTL, NAMECOM_HTTP_TIMEOUT
	netcup:		NETCUP_POLLING_INTERVAL, NETCUP_PROPAGATION_TIMEOUT, NETCUP_TTL, NETCUP_HTTP_TIMEOUT
	nifcloud:	NIFCLOUD_POLLING_INTERVAL, NIFCLOUD_PROPAGATION_TIMEOUT, NIFCLOUD_TTL, NIFCLOUD_HTTP_TIMEOUT
	ns1:		NS1_POLLING_INTERVAL, NS1_PROPAGATION_TIMEOUT, NS1_TTL, NS1_HTTP_TIMEOUT
	otc:		OTC_POLLING_INTERVAL, OTC_PROPAGATION_TIMEOUT, OTC_TTL, OTC_HTTP_TIMEOUT
	ovh:		OVH_POLLING_INTERVAL, OVH_PROPAGATION_TIMEOUT, OVH_TTL, OVH_HTTP_TIMEOUT
	pdns:		PDNS_POLLING_INTERVAL, PDNS_PROPAGATION_TIMEOUT, PDNS_TTL, PDNS_HTTP_TIMEOUT
	rackspace:	RACKSPACE_POLLING_INTERVAL, RACKSPACE_PROPAGATION_TIMEOUT, RACKSPACE_TTL, RACKSPACE_HTTP_TIMEOUT
	rfc2136:	RFC2136_POLLING_INTERVAL, RFC2136_PROPAGATION_TIMEOUT, RFC2136_TTL
	route53:	AWS_POLLING_INTERVAL, AWS_PROPAGATION_TIMEOUT, AWS_TTL
	sakuracloud:	SAKURACLOUD_POLLING_INTERVAL, SAKURACLOUD_PROPAGATION_TIMEOUT, SAKURACLOUD_TTL
	vegadns:	VEGADNS_POLLING_INTERVAL, VEGADNS_PROPAGATION_TIMEOUT, VEGADNS_TTL
	vultr:		VULTR_POLLING_INTERVAL, VULTR_PROPAGATION_TIMEOUT, VULTR_TTL, VULTR_HTTP_TIMEOUT

For a more detailed explanation of a DNS provider's credential variables,
please consult their online documentation.
```

### CODE REVIEW

File: cli.go [hunk 1]
```diff
@@ -204,42 +204,87 @@ Here is an example bash command using the CloudFlare DNS provider:
...
+	fmt.Fprintln(w, "\totc:\tOTC_POLLING_INTERVAL, OTC_PROPAGATION_TIMEOUT, OTC_TTL, OTC_HTTP_TIMEOUT")
+	fmt.Fprintln(w, "\tovh:\tOVH_POLLING_INTERVAL, OVH_PROPAGATION_TIMEOUT, OVH_TTL, OVH_HTTP_TIMEOUT")
+	fmt.Fprintln(w, "\tpdns:\tPDNS_POLLING_INTERVAL, PDNS_PROPAGATION_TIMEOUT, PDNS_TTL, PDNS_HTTP_TIMEOUT")
+	fmt.Fprintln(w, "\trackspace:\tRACKSPACE_POLLING_INTERVAL, RACKSPACE_PROPAGATION_TIMEOUT, RACKSPACE_TTL, RACKSPACE_HTTP_TIMEOUT")
+	fmt.Fprintln(w, "\trfc2136:\tRFC2136_POLLING_INTERVAL, RFC2136_PROPAGATION_TIMEOUT, RFC2136_TTL")
+	fmt.Fprintln(w, "\troute53:\tAWS_POLLING_INTERVAL, AWS_PROPAGATION_TIMEOUT, AWS_TTL")
+	fmt.Fprintln(w, "\tsakuracloud:\tSAKURACLOUD_POLLING_INTERVAL, SAKURACLOUD_PROPAGATION_TIMEOUT, SAKURACLOUD_TTL")
+	fmt.Fprintln(w, "\tvegadns:\tVEGADNS_POLLING_INTERVAL, VEGADNS_PROPAGATION_TIMEOUT, VEGADNS_TTL")
+	fmt.Fprintln(w, "\tvultr:\tVULTR_POLLING_INTERVAL, VULTR_PROPAGATION_TIMEOUT, VULTR_TTL, VULTR_HTTP_TIMEOUT")
+
```
Comment (wyattjoh) [inline cli.go]: Did not know that this was a thing.. Is it possible to move these variables to exported variables on each of the packages? That way it's a bit more safe?

File: providers/dns/dns_providers.go [hunk 2]
```diff
@@ -3,6 +3,11 @@ package dns
 import (
 	"fmt"
 
+	"github.com/xenolf/lego/providers/dns/ns1"
+	"github.com/xenolf/lego/providers/dns/otc"
+	"github.com/xenolf/lego/providers/dns/ovh"
+	"github.com/xenolf/lego/providers/dns/pdns"
+
```
Comment (wyattjoh) [inline providers/dns/dns_providers.go]: Remove extra space so it can order these imports.

File: cli.go (hunk 1 above)
Comment (ldez) [reply cli.go]: I am thinking about a system to better manage DNS provider constants and the display of help. I would like to do it in another PR.

Comment (ldez) [reply cli.go]: ping  @wyattjoh :smiley_cat:

Comment (wyattjoh) [reply cli.go]: Missed this comment, thank you!!

**Pull request merged**